### PR TITLE
sink-to-mysql(cdc) simplify conflict detector, prevent workload skew issue

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/10341

### What is changed and how it works?

- Remove early conflict resolving optimization for the conflict detector, prevent workload skew issue described in https://github.com/pingcap/tiflow/issues/10341
- Simplify the whole code logic, make it more easy to maintain.

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: pkg/causality/internal/node.go [hunk 1]
```diff
@@ -212,96 +180,41 @@ func (n *Node) assignTo(workerID int64) bool {
 	defer n.mu.Unlock()
 
 	if n.assignedTo != unassigned {
-		// Already resolved by some other guys.
+		// Already handled by some other guys.
```
Comment (asddongmen) [inline pkg/causality/internal/node.go:183]: // Already assign to a worker by another goroutrine.

**Review by asddongmen — APPROVED** [review]

File: pkg/causality/internal/node.go [hunk 2]
```diff
@@ -212,96 +180,41 @@ func (n *Node) assignTo(workerID int64) bool {
...
-// tryResolve try to find a worker to assign the node to.
-// Returns (_, false) if there is a conflict,
-// returns (rand, true) if there is no conflict,
-// returns (N, true) if only worker N can be used.
-func (n *Node) tryResolve(resolvedDependencies, removedDependencies int32) (int64, bool) {
-	assignedTo, resolved := n.doResolve(resolvedDependencies, removedDependencies)
-	if resolved && assignedTo == assignedToAny {
-		assignedTo = n.RandWorkerID()
+func (n *Node) maybeReadyToRun() {
+	if ok := n.checkReadiness(); ok {
```
Comment (CharlesCheung96) [inline pkg/causality/internal/node.go:197]: When a large number of small transactions are in conflict, does performance degrade significantly?

Comment (zhangjinpeng1987) [reply pkg/causality/internal/node.go:197]: It is suppose not fast in the case of there are a large number of small transactions have conflict, no matter what you do.

File: pkg/causality/internal/node.go (hunk 1 above)
Comment (zhangjinpeng1987) [reply pkg/causality/internal/node.go:183]: `handled` is a generic word here, not equal to `executed`.

**Review by hicqu — APPROVED** [review]

**Pull request merged**